### PR TITLE
Add cockroach tenet to check for discarded errors

### PR DESCRIPTION
**Pull request by Daanikus**

### CODE REVIEW

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - name: hasErrorComment
+    type: asserter
+    body: |
+      function(comment) {
+        return comment.toLowerCase().indexOf("error") !== -1
+      }
+tenets:
+  - name: do-not-discard-errors
+    flows:
+      codelingo/review:
```
Comment (waigani) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: There is no doc flow?

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        comment: |
+          Do not discard errors using _ variables. If a function returns an error,
```
Comment (waigani) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: This comment is better suited as a doc

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+funcs:
+  - name: hasErrorComment
```
Comment (mullikine) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: This function appears to not be used. Remove if you no longer need it

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml (hunk 1 above)
Comment (mullikine) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: misisng doc flow

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+          Do not discard errors using _ variables
+    query: |
+      import codelingo/ast/go
+
+      # This tenet should also check if the error
+      # discarding is mentioned in a comment above.
+      # See issues.yaml for more info
+
+      @review comment
+      go.block_stmt(depth = any):
```
Comment (waigani) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: Check for a comment containing the word "error".

File: tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    flows:
+      codelingo/docs:
+        title: Do Not Discard errors
+        body: |
+          Do not discard errors using _ variables. If a function returns an error, 
+          check it to make sure the function succeeded. Handle the error, return it, or, 
+          in truly exceptional situations, panic.
+      codelingo/review:
+        comment: |
+          Do not discard errors using _ variables
```
Comment (waigani) [inline tenets/codelingo/code-review-comments/do-not-discard-errors/codelingo.yaml]: Finish with a full stop. If ignoring the err is intentional, add a justification in a comment.

**Pull request merged**